Cargo.toml

```diff
@@ -67,6 +67,7 @@ opentelemetry-otlp = "0.28.0"
 opentelemetry_sdk =  "0.28.0"
 tracing-opentelemetry = "0.29.0"
 rayon = "1.10.0"
+async-lock = "3.4.0"
```

Comment:
While anyone could import this directly (and use it), I think it is helpful if we collect patterns that work with any `runtime` (see https://github.com/commonwarexyz/monorepo/issues/520#issuecomment-2770430117).